ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/ConfigureAmbariIdentitiesServerAction.java

```diff
@@ -126,10 +126,8 @@ protected CommandReport processIdentity(ResolvedKerberosPrincipal resolvedPrinci
       String hostName = resolvedPrincipal.getHostName();
       for (Map.Entry<String, String> serviceMappingEntry : resolvedPrincipal.getServiceMapping().entries()){
         String serviceName = serviceMappingEntry.getKey();
-        // distribute ambari keytabs only if host id is null, otherwise they will
-        // be distributed by usual process using ambari-agent.
         // TODO check if changes needed for multiple principals in one keytab
-        if (resolvedPrincipal.getHostId() == null && hostName != null && serviceName.equals(RootService.AMBARI.name())) {
+        if (hostName != null && serviceName.equals(RootService.AMBARI.name())) {
```

Comment:
This check should be to ensure the hostname is the same as the Ambari server's hostname.  This is being enforced below anyways:

```
if (StageUtils.getHostName().equals(hostName) && serviceName.equals(RootService.AMBARI.name()))
```

The following code is being used below this, forcing hostname to be the Ambari server's hostname:
```
          hostName = StageUtils.getHostName();
          File hostDirectory = new File(dataDirectory, hostName);
```

Reply:
Done.
Re-ran the same tests I described above, the result is the same:
```
[root@c7401 ~]# cat /etc/ambari-server/conf/krb5JAASLogin.conf
com.sun.security.jgss.krb5.initiate {
    com.sun.security.auth.module.Krb5LoginModule required
    renewTGT=false
    doNotPrompt=true
    useKeyTab=true
    keyTab="/etc/security/keytabs/ambari.server.keytab"
    principal="ambari-server-cluster1@AMBARI.APACHE.ORG"
    storeKey=true
    useTicketCache=false;
};
```